controllers/policy_controller.go

```diff
@@ -137,10 +138,26 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 				statusWriter, err := UpdateAWSObject(iamsvc, ins, DoNothingPreFunc)
 				statusWriter(ins, &policy, ctx, r.Status(), log)
 				if err != nil {
+					if aerr, ok := err.(awserr.Error); ok {
```

Comment:
This block of code will remove oldest policy if we are receiving "LimitExceeded" and will reconcile because spec.generation != status.observedGeneration